src/StreamJsonRpc/LengthHeaderMessageHandler.cs

```diff
@@ -32,7 +32,7 @@ public class LengthHeaderMessageHandler : PipeMessageHandler
         /// <summary>
         /// A wrapper to use for the <see cref="PipeMessageHandler.Writer"/> when we need to count bytes written.
         /// </summary>
-        private CountingWriterWrapper<byte> writerWrapper;
+        private PrefixingBufferWriter<byte> prefixingWriter;
```

Comment:
Neat! So this reserves some amount of space at the start that you can write later?

Reply:
Yes. And it's optimized so that no extra buffer copying is required, provided the payload that follows the header doesn't exceed a predicted length (otherwise 1 copy of the payload that exceeds that prediction is required).

Reply:
And David Fowler has expressed an interest in me contributing this class to corefx.